### PR TITLE
Fix group reimport bug

### LINKED ISSUE #54817: Godot 4: impossible to restart after creating a TextureAtlas from PNGs
### Godot version

v4.0.dev.20211108.official.596865366

### System information

Windows 10 64bit - Vulkan API 1.2.141 - GeForce GTX 1660 SUPER

### Issue description

When you create a `TextureAtlas ` from PNGs Godot as you to restart and can not start anymore without remove PNGs `.import` files.

I didn't find any related issue in the past months on GitHub tracker. I only tried it on Windows. I thing it's the same with other images. Same behavior in Godot_v4.0-dev.20211027.

### Steps to reproduce

1. Create a blank project
2. Add PNGs (I used https://www.kenney.nl/assets/toon-characters-1)
3. Select PNG in editor (Screenshot 1)
4. Select "Import" tab on left sidebar
5. Import as `TextureAtlas`
6. Select "Atlas file" and give it a name (the name does not matter)
7. Select "Reimport" button
8. Select "Save Scenes, Re-Import, and Restart" button
9. Wait the engine restart and display a blank pop-up named "Load Errors" (Screenshot 2)

image

image

### Minimal reproduction project

[texture-atlas_before.zip](https://github.com/godotengine/godot/files/7508066/texture-atlas_before.zip)

[texture-atlas_after.zip](https://github.com/godotengine/godot/files/7508067/texture-atlas_after.zip)

### LINKED ISSUE #68256: Creating AtlasTextures in Godot 4.0 beta 4 causes re-imports every time we reload, and engine runs slower
### Godot version

4.0 beta 4

### System information

Windows 11

### Issue description

I've started using AtlasTextures on all my sprites for the massive VRAM savings. 

There are three issues -: 
1) Everytime I open the project now, I see errors like this that eventually go away: 
image

2) There are reimports every time I open the project

3) The engine seems to be running on a different setting now. It is generally slower, and specially if I alt tab, it takes it a few seconds to wake up. This happens any time I change the window, and otherwise when navigating the filesystem. 

The VRAM savings are good, but slowed development is bad. Thanks for your help. 

### Steps to reproduce

1) Use atlastextures in project

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by AThousandShips**
When testing files for reimport the presence of a UID causes the file to be reimported (treated as if the file were from a version prior to UIDs I assume), this however causes problem with group files, in this case TextureAtlas, as in this case no UID is written (or kept in fact) when reimporting, causing the files in a group to be reimported each time

Also at some times (when opening the editor I believe) the resulting file (the group file) can be reimported itself causing it to be reimported twice in such cases, added a check to make sure files that are group files, but not yet in the group file cache, aren't reimported twice.

Edit: Also added editor progress update for group files in reimport

Fixes #68256
Fixes #54817

### CODE REVIEW

**Comment by kleonc** [comment]
I'd guess it also fixes #54817 which is supposed to be fixed by #60888 which... is almost identical to this PR. :smile:

**Comment by AThousandShips** [comment]
Oh damn I missed that one!

File: editor/editor_file_system.cpp [hunk 1]
```diff
@@ -2112,61 +2139,66 @@ void EditorFileSystem::reimport_files(const Vector<String> &p_files) {
...
-					do {
-						if (current_index < tdata.max_index) {
-							current_index = tdata.max_index;
-							pr.step(reimport_files[current_index].path.get_file(), current_index);
-						}
-						OS::get_singleton()->delay_usec(1);
-					} while (!WorkerThreadPool::get_singleton()->is_group_task_completed(group_task));
-
-					WorkerThreadPool::get_singleton()->wait_for_group_task_completion(group_task);
+		if (!groups_to_reimport.has(reimport_files[i].path)) {
```
Comment (kleonc) [inline editor/editor_file_system.cpp:2142]: Avoid indenting everything like that, instead a negated condition and early `continue`/`break`/`return` is preferred (just like in #60888).

**Comment by AThousandShips** [comment]
Think there are some factors the other PR doesn't cover so I'll leave this up for now unless you want me to

Thank you for finding

**Comment by kleonc** [comment]
> Think there are some factors the other PR doesn't cover so I'll leave this up for now unless you want me to close it

 Let @akien-mga decide what to do about this PR and #60888. Either way I think the changes need a review from @reduz.

**Comment by LakshayaG73** [comment]
Hi @akien-mga can we get some traction on this PR? 

It unblocks the basic scenarios of atlas textures which are very important for any 2D game.

**Comment by AThousandShips** [comment]
Also added a fix to ensure group files are updated properly after reimport, previously they would only be imported if the group file itself was reimported

**Comment by LakshayaG73** [comment]
After your latest commit, I deleted the .godot folder, and that seemed to error out some existing atlas import files. Not all, but some. 

Not sure what the connection is.

**Comment by AThousandShips** [comment]
Any reason you deleted the .godot folder? It would error stuff if it is not initialized again probably, as some of the parameters have changed if you are using both this and #68380, not sure what you are asking about or what the issue is.

**Comment by LakshayaG73** [comment]
I always delete the .godot folder after a fresh build. Is that not good practice? 

I am not using #68380 with this, for issues I have mentioned in the PR. 

I just thought I should mention this, in case you have any insight into the reason for this. It only messed with one specific file, so it could be simply due to the deletion of the folder. 

Otherwise, everything is working well.

**Comment by AThousandShips** [comment]
Got it! Suspect it's just because of the deletion if it works well after

**Review by Faless — APPROVED** [review]
LGTM, I also confirmed if fixes the two linked issues.
Really great work :trophy: ! Thanks!

**Comment by akien-mga** [comment]
Thanks!

**Comment by LakshayaG73** [comment]
Thank you again :)

**Comment by kleonc** [comment]
@akien-mga  I guess #60888 can be closed then (see https://github.com/godotengine/godot/pull/68324#issuecomment-1304852353).

**Comment by AThousandShips** [comment]
> Really great work
Thank you! Sorry forgot to reply before

**Pull request merged**